pubspec.yaml

```diff
@@ -1,12 +1,16 @@
 name: http_throttle
-version: 1.0.2
+version: 1.0.3
+
+description: HTTP client middleware that throttles requests.
 author: Dart Team <misc@dartlang.org>
 homepage: https://github.com/dart-lang/http_throttle
-description: HTTP client middleware that throttles requests.
+
 environment:
-  sdk: ">=2.0.0-dev.17.0 <2.0.0"
+  sdk: '>=2.0.0-dev.17.0 <3.0.0'
+
 dependencies:
-  http: ">=0.9.0 <0.12.0"
-  pool: ">=1.0.0 <2.0.0"
+  http: '>=0.9.0 <0.12.0'
+  pool: '>=1.0.0 <2.0.0'
+
 dev_dependencies:
-  test: ">=0.12.0 <0.13.0"
+  test: ^1.2.0
```

Comment:
Why are you narrowing this constraint rather than making it `>=0.12.0 <2.0.0`?

Reply:
Because:

- This is a dev_dependency.
- `test` is the _sole_ dev_dependency

So, AFAIK, in this case:

- No package can conflict with the version constraint of a dev_dependency (other than another dev_dependency, but there are none).
- In that case `pub get` will get at least latest version of `test`

So, we might as well just set the version of test to the current version (oh, which is 1.3.0).

Maybe my reasoning  is wrong?

Reply:
It's generally a good practice to leave the dependency as wide as possible while still guaranteeing that all the APIs you use are available, to give the version solver as much flexibility as possible.

Reply:
Ok. `0.12.35` is the fist Dart 2 compatible version. Since this package is now for Dart 2 only, shall I use `0.12.35` as the lowe rbound, or do you still prefer 0.12.0?

Reply:
Pub's version solver will sort out SDK compatibility. The pubspec only needs to concern itself with the APIs it uses, so I'd leave it at 0.12.0.

Reply:
Done